pkg/kube/kclient/clienttest/direct.go

```diff
@@ -79,7 +85,9 @@ func NewWriter[T controllers.ComparableObject](t test.Failer, c kube.Client) Tes
 // Typically, clienttest.WrapReadWriter should be used to simply wrap an existing client when testing an informer.
 // However, NewDirectClient can be useful if we do not need/want an informer and need direct reads.
 // Generic parameters represent the type with and without a pointer, and the list type.
-// Example: NewDirectClient[*Pod, Pod, PodList]
+// Example: NewDirectClient[*Pod, Pod, *PodList]
+// Istio types use a non-standard pointer type for lists. As a result, you need to set the pointer type twice.
+// Example: NewDirectClient[*ServiceEntry, *ServiceEntry, *ServiceEntryList](t, kc)
 func NewDirectClient[T controllers.ComparableObject, PT any, TL runtime.Object](t test.Failer, c kube.Client) TestClient[T] {
```

Comment:
I guess you could do something similar with `NewDirectClient(isIstioType bool)` and ditch the weird type constraint stuff but that's mostly different-not-better.